File: backend/src/main/java/bg/pcbudget/backend/models/GPU.java
package bg.pcbudget.backend.models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "gpus")
@Getter
@Setter
@NoArgsConstructor
public class GPU extends Product {

  private String chipBrand;
  private String graphicsProcessor;

  @ManyToOne
  @JoinColumn(name = "interface_id")
  private ItemInterface interfaceType;

  private Integer memorySizeGB;
  private String memoryType;
  private Integer slotWidth;
  private String directXVersion;
}

File: src/app/models/GPU.ts
import { Product } from './Product';
import { Interface } from './Interface';

export interface GPU extends Product {
  chipBrand:         string;
  graphicsProcessor: string;
  interfaceType:     Interface;
  memorySizeGB:      number;
  memoryType:        string;
  slotWidth:         number;
  directXVersion:    string;
}
